experanto/dataloaders.py

```diff
@@ -1,17 +1,131 @@
+            If a list is provided, it should match the length of paths.
+        shuffle_keys (bool): Whether to shuffle the keys of the dataloaders.
+        **kwargs: Additional keyword arguments for dataset and dataloader configuration.
+    """
+
+    if configs is None and "config" in kwargs:
+        configs = kwargs.pop("config")
+
+    # Convert single config to list for uniform handling
+    if isinstance(configs, (DictConfig, dict)):
+        configs = [configs] * len(paths)
 
 
-def get_multisession_dataloader(paths, config: DictConfig) -> DataLoader:
     dataloaders = {}
-    for i, path in enumerate(paths):
-        dataset_name = path.split("dynamic")[1].split("-Video")[0]
-        dataset = ChunkDataset(path, **config.dataset,)
+    for i, (path, cfg) in enumerate(zip(paths, configs)):
+        # TODO use saved meta dict to find data key
+        if "dynamic" in path:
+            dataset_name = path.split("dynamic")[1].split("-Video")[0]
+        elif "_gaze" in path:
+            dataset_name = path.split("_gaze")[0].split("datasets/")[1]
+        else:
+            dataset_name = f"session_{i}"
+        dataset = ChunkDataset(path, **cfg.dataset)
         dataloaders[dataset_name] = MultiEpochsDataLoader(dataset,
-                                               **config.dataloader,
-                                               )
-    return LongCycler(dataloaders)
\ No newline at end of file
+                                                          **cfg.dataloader,)
+
+
+
+    return LongCycler(dataloaders)
+
+
+def get_multisession_concat_dataloader(paths: List[str],
+                                       configs: Union[Dict, List[Dict]] = None,
+                                       seed: Optional[int] = 0,
+                                       **kwargs) -> 'FastSessionDataLoader':
+    """
+    Creates a multi-session dataloader using SessionConcatDataset and SessionDataLoader.
+    Returns (session_key, batch) pairs during iteration.
+
+    Args:
+        paths: List of paths to dataset files
+        configs: Configuration for datasets (single config or list of configs)
+        seed: Random seed for reproducibility
+        num_workers: Number of worker processes for data loading
+        prefetch_factor: Prefetch factor for data loading
+        **kwargs: Additional arguments
+
+    Returns:
+        SessionDataLoader instance or None if no valid datasets found
+    """
+    if configs is None and "config" in kwargs:
+        configs = kwargs.pop("config")
+
+    # Convert single config to list for uniform handling
+    if not isinstance(configs, list):
+        configs = [configs] * len(paths)
+
+    # Create datasets
+    datasets = []
+    session_names = []
+
+    start_time = time.time()
+    for i, (path, cfg) in enumerate(zip(paths, configs)):
+        # Extract session name
+        if "dynamic" in path:
+            session_name = path.split("dynamic")[1].split("-Video")[0]
```

Comment:
Same here + duplicate code.

Reply:
Agreed. I left both `dataloaders` convenience functions in, mostly for comparison and benchmarking.

(1) `get_multisession_dataloader` is using the LongCyler logic.
(2) `get_multisession_concat_dataloader` is using the torch.ConcatDataset logic.

Happy to get rid of (1) entirely, and with that all the cycler/Longcycler/Exhauster business.